avocado/plugins/run.py

```diff
@@ -185,4 +186,13 @@ def run(self, args):
             log.error(e.message)
             sys.exit(exit_codes.AVOCADO_FAIL)
         job_instance = job.Job(args)
-        return job_instance.run()
+        job_run = job_instance.run()
+        result_dispatcher = ResultDispatcher()
+        if result_dispatcher.extensions:
+            # At this point job_instance doesn't have a single results attribute
+            # which is the end goal.  For now, we pick any of the plugin classes
+            # added to the result proxy.
+            if len(job_instance.result_proxy.output_plugins) > 0:
+                result = job_instance.result_proxy.output_plugins[0]
+                result_dispatcher.map_method('render', result, job_instance)
```

Comment:
The idea is indeed to have a single copy of `Result` *it's mentioned in one comment IIRC).  `ResultProxy` is on its way to being removed, in favor of the "new style" dispatchers.  I have the feeling that a proper (different) plugin interface for "streamable/events based results" is the way to go.  So plugin writers can choose if their result plugins are simple "Result" plugins or "Streamable/Events based Result" plugins.

But, it's probably going to be better discussed with supporting code in an future PR.